### PR TITLE
Automate API stubbed tests in test_role.py

### PR DESCRIPTION
**Pull request by shubhamsg199**
**`Test Results:`**
```
$ pytest -k test_negative_access_entities_from_ldap_org_admin tests/foreman/api/test_role.py 
====================================== test session starts ===================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sganar/satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.6.1, forked-1.3.0, ibutsu-2.0.2, reportportal-5.0.11, cov-3.0.0, xdist-2.5.0
collected 50 items / 49 deselected / 1 selected                                                                                                                                                                                                                                          

tests/foreman/api/test_role.py .                                                                                                                                                                                                                                                   [100%]

=============================== 1 passed, 49 deselected in 88.64s (0:01:28) ==========================================================
```

```
$ pytest -k test_negative_access_entities_from_ldap_user tests/foreman/api/test_role.py 
=========================================== test session starts ===============================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sganar/satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.6.1, forked-1.3.0, ibutsu-2.0.2, reportportal-5.0.11, cov-3.0.0, xdist-2.5.0
collected 50 items / 49 deselected / 1 selected                                                                                                                                                                                                                                          

tests/foreman/api/test_role.py .                                                                                                                                                                                                                                                   [100%]

================================= 1 passed, 49 deselected in 88.79s (0:01:28) ====================================================
```
```
$ pytest -k test_positive_assign_org_admin_to_ldap_user_group tests/foreman/api/test_role.py 
======================================== test session starts ==================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sganar/satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.6.1, forked-1.3.0, ibutsu-2.0.2, reportportal-5.0.11, cov-3.0.0, xdist-2.5.0
collected 50 items / 49 deselected / 1 selected                                                                                                                                                                                                                                          

tests/foreman/api/test_role.py .                                                                                                                                                                                                                                                   [100%]

================================= 1 passed, 49 deselected in 85.77s (0:01:25) =======================================================
```
```
$ pytest -k test_negative_assign_org_admin_to_ldap_user_group tests/foreman/api/test_role.py 
========================================= test session starts ========================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sganar/satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.6.1, forked-1.3.0, ibutsu-2.0.2, reportportal-5.0.11, cov-3.0.0, xdist-2.5.0
collected 50 items / 49 deselected / 1 selected                                                                                                                                                                                                                                          

tests/foreman/api/test_role.py .                                                                                                                                                                                                                                                   [100%]

===================================== 1 passed, 49 deselected in 84.95s (0:01:24) ===================================
```

### CODE REVIEW

File: tests/foreman/api/test_role.py [hunk 1]
```diff
@@ -1582,12 +1618,35 @@ def test_negative_access_entities_from_ldap_org_admin(self):
...
+        )
+        sc = ServerConfig(
+            auth=(create_ldap['ldap_user_name'], create_ldap['ldap_user_passwd']),
+            url=create_ldap['sat_url'],
+            verify=False,
+        )
+        with pytest.raises(HTTPError):
+            entities.Architecture(sc).search()
+        user = entities.User().search(
+            query={'search': 'login={}'.format(create_ldap['ldap_user_name'])}
```
Comment (mshriver) [inline tests/foreman/api/test_role.py:1638]: Please use f-strings for string formatting.

**Review by mshriver — APPROVED** [review]

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: -k 'test_negative_access_entities_from_ldap_ or test_negative_assign_org_admin_to_ldap_ or test_positive_assign_org_admin_to_ldap_' tests/foreman/api/test_role.py

**Comment by shubhamsg199** [comment]
trigger: test-robottelo
pytest: -k 'test_negative_access_entities_from_ldap_ or test_negative_assign_org_admin_to_ldap_ or test_positive_assign_org_admin_to_ldap_' tests/foreman/api/test_role.py

File: tests/foreman/api/test_role.py [hunk 2]
```diff
@@ -1631,16 +1705,49 @@ def test_positive_assign_org_admin_to_ldap_user_group(self):
...
+        ).create()
+
+        for user in users:
+            sc = ServerConfig(
+                auth=(user.login, password),
+                url=create_ldap['sat_url'],
+                verify=False,
+            )
+            # Accessing the Domain resource
+            entities.Domain(sc, id=domain.id).read()
```
Comment (pondrejk) [inline tests/foreman/api/test_role.py:1743]: No wrapping into `with pytest.raises(HTTPError):` here?

**Review by pondrejk — CHANGES_REQUESTED** [review]

File: tests/foreman/api/test_role.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/foreman/api/test_role.py:1743]: Not required here as this is a positive test scenario and the domain resource we are accessing here is in the same taxonomy as the org_admin_role, hence the user should be able to access the Domain resource created.

**Review by pondrejk — APPROVED** [review]

**Pull request merged**